instance/tests/test_tasks.py

```diff
@@ -75,8 +77,20 @@ def test_mark_active_on_success(self, provisioning_succeeds):
         instance = OpenEdXInstanceFactory()
         tasks.spawn_appserver(instance.ref.pk, mark_active_on_success=True)
         self.assertEqual(self.mock_spawn_appserver.call_count, 1)
+        self.assertEqual(self.mock_make_appserver_active.call_count, 1 if provisioning_succeeds else 0)
 
-        self.assertEqual(self.mock_add_active_appserver.call_count, 1 if provisioning_succeeds else 0)
+    @ddt.data(True, False)
+    def test_deactivate_old_appservers(self, provisioning_succeeds):
+        """
+        If `mark_active_on_success` and `deactivate_old_appservers` are both passed in as `True`,
+        the spawn appserver task will mark the newly provisioned AppServer as active, and deactivate
+        old appservers, if provisioning succeeded.
+        """
+        self.mock_spawn_appserver.return_value = 10 if provisioning_succeeds else None
+        instance = OpenEdXInstanceFactory()
+        tasks.spawn_appserver(instance.ref.pk, mark_active_on_success=True, deactivate_old_appservers=True)
+        self.assertEqual(self.mock_spawn_appserver.call_count, 1)
+        self.assertEqual(self.mock_make_appserver_active.call_count, 1 if provisioning_succeeds else 0)
```

Comment:
This test doesn't seem to actually check if the old appservers got deactivated? Or even just if `deactivate_others=True` got passed to `make_appserver_active`?

(I know `test_deactivate_others` does, but it's a more narrow unit test)

Reply:
I've changed it to test that the right arguments are passed in. :+1: